### PR TITLE
Add container mutation tests for inserting functions into dictionaries

### PR DESCRIPTION
**Pull request by turbolent**
## Description

Just like for arrays, check the insertion of various kinds of functions into dictionaries.

This was triggered by a report from @sideninja that insertion of functions into dictionaries is not working on Mainnet.

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/tests/interpreter/container_mutation_test.go [hunk 1]
```diff
@@ -650,6 +648,221 @@ func TestDictionaryMutation(t *testing.T) {
...
+	t.Run("valid function mutation", func(t *testing.T) {
+		t.Parallel()
+
+		inter := parseCheckAndInterpret(t, `
+            struct S {}
+
+	        fun test(owner: PublicAccount) {
+	            let funcs: {String: ((PublicAccount, [UInt64]): [S])} = {}
+
+                funcs["test"] = fun (owner: PublicAccount, ids: [UInt64]): [S] { return [] }
```
Comment (dsainati1) [inline runtime/tests/interpreter/container_mutation_test.go:852]:
```suggestion
                  funcs["test"] = fun (owner: PublicAccount, ids: [UInt64]): [S] { return [] }
```

File: runtime/tests/interpreter/container_mutation_test.go [hunk 2]
```diff
@@ -650,6 +648,221 @@ func TestDictionaryMutation(t *testing.T) {
...
+		assert.Equal(t, sema.VoidType, funcType.ReturnTypeAnnotation.Type)
+		assert.Nil(t, funcType.ReceiverType)
+		assert.Empty(t, funcType.Parameters)
+	})
+
+	t.Run("valid function mutation", func(t *testing.T) {
+		t.Parallel()
+
+		inter := parseCheckAndInterpret(t, `
+            struct S {}
```
Comment (dsainati1) [inline runtime/tests/interpreter/container_mutation_test.go:847]:
```suggestion
              struct S {}
```

Comment (turbolent) [reply runtime/tests/interpreter/container_mutation_test.go:847]: Cleaned up the whole file in 6e064c5a77fffe70d3db7c2b9035062348b0ac92

File: runtime/tests/interpreter/container_mutation_test.go (hunk 1 above)
Comment (turbolent) [reply runtime/tests/interpreter/container_mutation_test.go:852]: Cleaned up the whole file in 6e064c5a77fffe70d3db7c2b9035062348b0ac92

**Review by SupunS — APPROVED** [review]
LGTM! Thank you for adding this!

**Pull request merged**